### PR TITLE
Add subclassing to reduce repeated code

### PR DESCRIPTION
**Pull request by fwkoch**

### CODE REVIEW

File: vectormath/vector.py [hunk 1]
```diff
@@ -6,28 +6,15 @@
...
-            if X is None and Y is None and Z is None:
-                return cls(0, 0, 0)
-            if np.isscalar(X) and np.isscalar(Y) and np.isscalar(Z):
-                xyz = np.r_[X, Y, Z]
-                xyz = xyz.astype(float)
-                return xyz.view(cls)
-            raise ValueError('Invalid input for Vector3 - must be an instance '
-                             'of a Vector3, a length-3 array, 3 scalars, or '
-                             'nothing for [0., 0., 0.]')
+class Vector(np.ndarray):
```
Comment (rowanc1) [inline vectormath/vector.py]: Can we call this `BaseVector`

File: vectormath/vector.py [hunk 2]
```diff
@@ -6,28 +6,15 @@
...
 
-        return read_array(x, y, z)
+    def __new__(cls, *args, **kwargs):
+        for vcls in (Vector3, Vector2):
```
Comment (rowanc1) [inline vectormath/vector.py]: Typo: `cls`?

File: vectormath/vector.py [hunk 3]
```diff
@@ -6,28 +6,15 @@
...
+            try:
+                return vcls(*args, **kwargs)
```
Comment (rowanc1) [inline vectormath/vector.py]: cls?

**Review by rowanc1 — CHANGES_REQUESTED** [review]
Looks good.

File: vectormath/vector.py (hunk 2 above)
Comment (fwkoch) [reply vectormath/vector.py]: That isn't a typo - constructing Vector (`cls`) tries to construct either Vector3 or Vector2 (the `vcls`es). I think I'll just take it out though... It is unnecessary complication. `BaseVector` and `BaseVectorArray` should just not be used at all.

**Pull request merged**